### PR TITLE
RFC: Replace hardcoded targets/patches paths with appropriate GLUON_ make variables

### PR DESCRIPTION
**Pull request by ynezz**
Proposed changes allows me to have have custom `patches` and `targets` in the site repository. As a bonus, it removes bunch of hardcoded paths, making the build system more flexible. I'm wondering if such changes would be accepted.

```
$ cat site/site.mk
...
GLUON_PATCHESDIR ?= $(GLUON_SITEDIR)/patches
GLUON_TARGETSDIR ?= $(GLUON_SITEDIR)/targets
...
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +27,7 @@ $(eval $(call mkabspath,GLUON_OUTPUTDIR))
 $(eval $(call mkabspath,GLUON_IMAGEDIR))
 $(eval $(call mkabspath,GLUON_PACKAGEDIR))
 
-export GLUON_TMPDIR GLUON_IMAGEDIR GLUON_PACKAGEDIR DEVICES
+export GLUON_TMPDIR GLUON_IMAGEDIR GLUON_PACKAGEDIR DEVICES V
```
Comment (mweinelt) [inline Makefile]: V?

**Comment by mweinelt** [comment]
The patchset carried within gluon is necessary to make alot of things actually work, so replacing it by something else is nothing I would recommend.

Being able to carry additional patches in the site repository however is a great idea.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (ynezz) [reply Makefile]: Indeed, I must have screwed something in here, because `make V=s` wasn't working for me somehow.

**Comment by ynezz** [comment]
> The patchset carried within gluon is necessary to make alot of things actually work, so replacing it by something else is nothing I would recommend.

Sure thing, but you can cherry-pick just those patches you want to use exactly.

> Being able to carry additional patches in the site repository however is a great idea.

I was thinking about this layers based approach as well, but it would make the stuff more complicated during update of patches(scripts/update-patches.sh) for example.

**Comment by christf** [comment]
@mweinelt it could take the edge off a restrictive patch-merging policy. It seems, if we had this it would be easier to run community-based patches to test stuff. Like the batman multicast patch for example.

**Comment by mweinelt** [comment]
Easy enough, fork gluon, merge your patch, build away. Then remember to rebase early and often.

Patch handling is described here: https://gluon.readthedocs.io/en/v2018.1.x/dev/basics.html

**Comment by ynezz** [comment]
> Easy enough, fork gluon, merge your patch, build away. Then remember to rebase early and often.

Those are exactly the steps I would like to avoid. With this small patchset I could just use upstream Gluon, with custom `site` repository where I've all the custom bits.

**Comment by rotanid** [comment]
@NeoRaider if you like it, do you still have issues with any details or can it be merged?

**Comment by NeoRaider** [comment]
I was still unsure if I like the ordering/aggregation of the variable exports. After thinking about this a bit more, I'd prefer to keep the different GLUON_*DIR variable definitions and exports together (and separate from the export for the non-*DIR variables).

My suggestion is to move the definition of all DIR variables except for GLUON_SITEDIR down below the site.mk import. White at it, we can also move the export of DEVICES to the other export directive (where GLUON_RELEASE etc. are exported).

**Comment by rotanid** [comment]
@ynezz do you plan to implement the requested changes?

**Comment by ynezz** [comment]
@rotanid Yes, I'll work on this next week.

**Comment by ynezz** [comment]
@NeoRaider Is it better now?

**Comment by NeoRaider** [comment]
Thanks, rebased and merged with minor adjustments.

**Pull request closed without merging**